descriptors/descriptors.go

```diff
@@ -4,27 +4,33 @@ import (
 	_ "embed"
 )
 
+type wasmInstance struct {
+	mod  *wasmModule
+	ptr  uint64
+	drop func()
+}
+
 type Descriptor struct {
-	mod   *wasmModule
-	ptr   uint64
-	close func()
+	wasmInstance
 }
```

Comment:
What's the purpose of this transformation? I am not a big fan of struct embedding, in my experience it tends to cause more trouble than is worth it. If your goal was to encapsulate the mod/ptr/drop parts, please make it a proper struct field like `wasmInstance wasmInstance`

Reply:
Had some additional refactoring in a previous version. But doesn't make sense anymore, so reverted this.